src/radare.cpp

```diff
@@ -23,12 +23,20 @@
 
 namespace lo2s
 {
-Radare::Radare() : r_asm_(r_asm_new())
+Radare::Radare() : r_lib_(r_lib_new(NULL, NULL)), r_anal_(r_anal_new()), r_asm_(r_asm_new())
 {
-    r_asm_use(r_asm_, "x86");
-    r_asm_set_bits(r_asm_, 64);
-    r_asm_set_syntax(r_asm_, R_ASM_SYNTAX_ATT);
-    r_asm_set_pc(r_asm_, 0x0);
+    r_unref(r_anal_->config);
+
+    r_asm_->num = r_num_new(NULL, NULL, NULL);
+    r_anal_->config = r_ref_ptr(r_asm_->config);
+    r_anal_bind(r_anal_, &r_asm_->analb);
+
+    r_asm_use(r_asm_, R_SYS_ARCH);
+    r_anal_use(r_anal_, R_SYS_ARCH);
+
+    int sysbits = (R_SYS_BITS & R_SYS_BITS_64) ? 64 : 32;
+    r_asm_set_bits(r_asm_, sysbits);
+    r_anal_set_bits(r_anal_, sysbits);
 }
 
 std::string Radare::single_instruction(char* buf)
```

Comment:
The method could also have some love.
`buf` might be null. And the code is a bit inefficient.

Maybe something like this:
```c++
   if (buf == nullptr) throw ...;
   auto it = buf;
   while(*it != '\0' && *it != '\n') {
      it++;
   }
   if (it == buf) throw Error("empty instruction");
    return std::string(buf, it - 1);
```